internal/pproflogging/pproflogging.go

```diff
@@ -97,10 +103,41 @@ var pprofProfileRates = map[ProfileName]pprofSetRate{
 	},
 }
 
+func newProfileConfigs(wrt Writer) *ProfileConfigs {
+	q := &ProfileConfigs{
+		wrt: wrt,
+	}
+
+	return q
+}
+
+// LoadProfileConfig configure PPROF profiling from the config in ppconfigss.
+func LoadProfileConfig(ctx context.Context, ppconfigss string) (map[ProfileName]*ProfileConfig, error) {
+	// if empty, then don't bother configuring but emit a log message - use might be expecting them to be configured
+	if ppconfigss == "" {
+		log(ctx).Debug("no profile configuration. skipping PPROF setup")
+		return nil, nil
+	}
+
+	bufSizeB := DefaultDebugProfileDumpBufferSizeB
+
+	// look for matching services.  "*" signals all services for profiling
+	log(ctx).Info("configuring profile buffers")
+
+	// acquire global lock when performing operations with global side-effects
```

Comment:
do you still need the global lock here since you are mostly doing in-memory work ? Maybe this comment is now obsolete ?

Reply:
Its to prevent PPROF dump output from overlapping in the log output.  If they get mixed in the output its impossible to distinguish which base-64 encoded line belongs to which profile.

Reply:
got it, thanks for the clarification